### ISSUE TITLE
Creating Polygon Layer - Color is wrong and cannot be changed

### ISSUE THREAD

**Issue by michaelc12143** [qgis/QGIS#57861]
### What is the bug or the crash?

I create a new layer for a polygon to create a lake on a map. Create the lake, but cannot apply the color (in this case #4baffb). Instead it comes up with a random color. Once in a great while I get the correct color. Once the layer is completed, I cannot go back later and correct the color. The original random color is permanent.

### Steps to reproduce the issue

Desktop running Windows 10
QGIS version is 3.36.3

1. Open Layer Panel on left side of screen (View > Panels > Layers)
2. Add Annotations to Toolbar (View > Toolbars > Annotations Toolbar)
3. In Layer Panel - Move cursor to one of the existing waterbodies and Click Once
4. Create Layer (Layer > Create Layer > New Shapefile Layer>
File Name (enter lake name)
Geometry Type (Polygon)
New Field (Name=name, Type=Text String, Length=”30” > Click Add toField List
New Field (Name=location, Type=Text String, Length=30 > Click Add Field
Click OK
5. Double Click on new lake in Layer Panel
Change Color to #4baffb
Click Apply & OK

6. On the Annotations Toolbar Click on 'Create Polygon Annotation' icon
Create Lake
7. Once lake is created, right click and apply color

**Problem #1**: Most of the time you get a random color like green, purple, red, etc. Once in a great while, you get the correct color.

**Problem #2:** If you click on the layer I just created, the correct color  (#4bafb) comes up, but if you try to apply the color, it will not change it. Cannot find a work-around to change the color.

Note: The color of older shape files of lakes created under 3.22 can be changed under 3.36.3. Just not new shape files created under 3.36.3

Could not find a work-around for these two problems.

### Versions

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0//EN" "http://www.w3.org/TR/REC-html40/strict.dtd">
<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" /><style type="text/css">
p, li { white-space: pre-wrap; }
</style></head><body>

QGIS version | 3.36.3-Maidenhead | QGIS code revision | 2df96554
-- | -- | -- | --
Qt version | 5.15.13
Python version | 3.12.3
GDAL/OGR version | 3.9.0
PROJ version | 9.4.0
EPSG Registry database version | v11.004 (2024-02-24)
GEOS version | 3.12.1-CAPI-1.18.1
SQLite version | 3.45.1
PDAL version | 2.6.3
PostgreSQL client version | 16.2
SpatiaLite version | 5.1.0
QWT version | 6.2.0
QScintilla2 version | 2.14.1
OS version | Windows 10 Version 2009
  |   |   |  
Active Python plugins
QuickOSM | 2.1.1
quick_map_services | 0.19.32
db_manager | 0.1.20
grassprovider | 2.12.99
MetaSearch | 0.3.6
processing | 2.12.99

</body></html>QGIS version
3.36.3-Maidenhead
QGIS code revision
[2df96554](https://github.com/qgis/QGIS/commit/2df96554)
Qt version
5.15.13
Python version
3.12.3
GDAL/OGR version
3.9.0
PROJ version
9.4.0
EPSG Registry database version
v11.004 (2024-02-24)
GEOS version
3.12.1-CAPI-1.18.1
SQLite version
3.45.1
PDAL version
2.6.3
PostgreSQL client version
16.2
SpatiaLite version
5.1.0
QWT version
6.2.0
QScintilla2 version
2.14.1
OS version
Windows 10 Version 2009

Active Python plugins
QuickOSM
2.1.1
quick_map_services
0.19.32
db_manager
0.1.20
grassprovider
2.12.99
MetaSearch
0.3.6
processing
2.12.99

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

**Comment by nicogodet** [comment]
Could you confirm you use the right tool to create the polygon ?
![image](https://github.com/qgis/QGIS/assets/39594821/89468906-1e50-4b24-8419-e8a9e8558124)

Create annotation polygon is a completly different tool which will not add a polygon entity to your shapefile layer.
See doc about annotation layer : https://docs.qgis.org/3.34/en/docs/user_manual/map_views/map_view.html#annotation-tools

**Comment by michaelc12143** [comment]
Apparently I am using the wrong one. But, the icon you are showing is not on my system.

**Comment by michaelc12143** [comment]
Okay I figured it out. Thank you!

**Comment by michaelc12143** [comment]
While that worked, I cannot remove the original layer. It does not show up in the panel as a layer, but is being displayed on the map.

**Comment by michaelc12143** [comment]
I was using the wrong icon to create the shape. 
Once I clicked on the pencil, I was able to see the proper icon to use.

The only problem I had was the original shape I 
created is not in the layer panel, but shows up 
on the map. Cannot figure out how to remove it. I 
have checked all the file directories, and 
searched the disk drive. But, still cannot find it.

Michael


At 10:29 AM 6/24/2024, you wrote:
